### PR TITLE
Save game

### PR DESCRIPTION
**Pull request by szeliga71**
simple class for saving game history.

### CODE REVIEW

File: src/main/java/org/example/gameService/SaveGame.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    private File file;
+    private String gameHistoryFilePath;
+    private DateTimeFormatter formatter;
+    private LocalTime time;
+
+    public SaveGame(GameData gameData) {
+
+        this.gameData = gameData;
+        this.date = LocalDate.now();
+        this.time = LocalTime.now();
```
Comment (blazejprzyluski) [inline src/main/java/org/example/gameService/SaveGame.java]: time and date is the same

File: src/main/java/org/example/gameService/VCSFileCreator.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        schema.put("playerPoints", 12);
+        schema.put("changeInPoints", 14);
+        schema.put("playerWin", 9);
+        schema.put("sesja z godziny", 26);
+        return schema;
+    }
+
+    public String createVCSContent(GameData gameData) {
+
+        if(gameData!=null){
```
Comment (blazejprzyluski) [inline src/main/java/org/example/gameService/VCSFileCreator.java]: ctrl + alt + l

File: src/main/java/org/example/gameService/VCSFileCreator.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        schema.put("houseScore", 10);
+        schema.put("playerScore", 11);
+        schema.put("playerPoints", 12);
+        schema.put("changeInPoints", 14);
+        schema.put("playerWin", 9);
+        schema.put("sesja z godziny", 26);
+        return schema;
+    }
+
+    public String createVCSContent(GameData gameData) {
```
Comment (blazejprzyluski) [inline src/main/java/org/example/gameService/VCSFileCreator.java:29]: could this method be static?

File: src/main/java/org/example/gameService/SaveGame.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+
+public class SaveGame {
+
+    private VCSFileCreator vcsFileCreator;
```
Comment (blazejprzyluski) [inline src/main/java/org/example/gameService/SaveGame.java]: some of those field seem unnecessary, as they're not used anywhere else in the class

File: src/test/java/SaveGameTest.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+@ExtendWith(MockitoExtension.class)
+public class SaveGameTest {
+
+
+    @Mock
+    private final GameData gameData = new GameData(1, 2, 3, 4, true);
+    @Mock
+    private final SaveGame saveGame = new SaveGame(gameData);
+    @Mock
+    VCSFileCreator fileCreator = new VCSFileCreator();
```
Comment (blazejprzyluski) [inline src/test/java/SaveGameTest.java]: private

File: src/test/java/SaveGameTest.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    @AfterEach
+    void cleanUp() {
+        deleteTemporaryDirectoryAndFiles(new File("src/main/TestHistory/"));
+    }
+
+    LocalDate date = LocalDate.now();
+    String gameHistoryFilePathTest = "src/main/TestHistory/test history " + date + ".csv";
+    Path path = Paths.get(gameHistoryFilePathTest);
+    File file = path.toFile();
+    File testFile = new File(gameHistoryFilePathTest);
```
Comment (blazejprzyluski) [inline src/test/java/SaveGameTest.java]: make all of them private

File: src/test/java/SaveGameTest.java [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+          if(files != null){
+              for(File f : files){
+                  deleteTemporaryDirectoryAndFiles(f);
+              }
+          }
+        }
+        file.delete();
+    }
+    @Test
+    void saveGameHistorNullArgument() {
```
Comment (blazejprzyluski) [inline src/test/java/SaveGameTest.java]: saveGameHistory

File: src/main/java/org/example/gameService/SaveGame.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    private GameData gameData;
+    private LocalDate date;
+    private Path path;
+    private File file;
+    private String gameHistoryFilePath;
+    private DateTimeFormatter formatter;
+    private LocalTime time;
+
+    public SaveGame(GameData gameData) {
```
Comment (blazejprzyluski) [inline src/main/java/org/example/gameService/SaveGame.java]: Why does SaveGame get gameData in the constructor, when the method responsible for saving is using other gameData?

File: src/test/java/VCSFileCreatorTest.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+import org.example.model.GameData;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+
+@ExtendWith(MockitoExtension.class)
+public class VCSFileCreatorTest {
```
Comment (blazejprzyluski) [inline src/test/java/VCSFileCreatorTest.java:17]: Please add more tests checking content/formatting of VCS file

File: src/test/java/VCSFileCreatorTest.java [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
+
+    @Mock
+    GameData localGameData = new GameData(1, 2, 3, 4, true);
```
Comment (blazejprzyluski) [inline src/test/java/VCSFileCreatorTest.java]: Wrong usage of Mock, this is not a mock, but a concrete instance of an object

File: src/test/java/SaveGameTest.java [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+
+import static org.mockito.Mockito.mock;
+
+
+@ExtendWith(MockitoExtension.class)
+public class SaveGameTest {
+
+
+    @Mock
+    private final GameData gameData = new GameData(1, 2, 3, 4, true);
```
Comment (blazejprzyluski) [inline src/test/java/SaveGameTest.java]: not a mock but real object

File: src/test/java/SaveGameTest.java [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+    @Mock
+    private final SaveGame saveGame = new SaveGame(gameData);
```
Comment (blazejprzyluski) [inline src/test/java/SaveGameTest.java]: not a mock but real object

File: src/test/java/SaveGameTest.java (hunk 5 above)
Comment (blazejprzyluski) [reply src/test/java/SaveGameTest.java]: not a mock but real object

File: src/main/java/org/example/gameService/SaveGame.java (hunk 1 above)
Comment (szeliga71) [reply src/main/java/org/example/gameService/SaveGame.java]: fixed

File: src/main/java/org/example/gameService/VCSFileCreator.java (hunk 2 above)
Comment (szeliga71) [reply src/main/java/org/example/gameService/VCSFileCreator.java]: Done

File: src/main/java/org/example/gameService/VCSFileCreator.java (hunk 3 above)
Comment (szeliga71) [reply src/main/java/org/example/gameService/VCSFileCreator.java:29]: Previously, I quckly created the VCSFileCreator class, which resulted in poor considered fields and poor structured methods in both the SaveGame and VCSFileCreator classes. In the current commit, I have cleaned up the code by removing unnecessary fields and improving the methods.

File: src/main/java/org/example/gameService/SaveGame.java (hunk 4 above)
Comment (szeliga71) [reply src/main/java/org/example/gameService/SaveGame.java]: I hope fixed. As above.

File: src/test/java/SaveGameTest.java (hunk 7 above)
Comment (szeliga71) [reply src/test/java/SaveGameTest.java]: I have modified test methods.

File: src/test/java/SaveGameTest.java (hunk 6 above)
Comment (szeliga71) [reply src/test/java/SaveGameTest.java]: fixed

File: src/test/java/SaveGameTest.java (hunk 5 above)
Comment (szeliga71) [reply src/test/java/SaveGameTest.java]: _I have modified tests methods._

File: src/main/java/org/example/gameService/SaveGame.java (hunk 8 above)
Comment (szeliga71) [reply src/main/java/org/example/gameService/SaveGame.java]: Previously, I quckly created the VCSFileCreator class from SaveGame class, which resulted in poor considered fields and poor structured methods in both the SaveGame and VCSFileCreator classes. In the current commit, I have cleaned up the code by removing unnecessary fields and improving the methods.

File: src/test/java/VCSFileCreatorTest.java (hunk 9 above)
Comment (szeliga71) [reply src/test/java/VCSFileCreatorTest.java:17]: Done

File: src/test/java/SaveGameTest.java (hunk 11 above)
Comment (szeliga71) [reply src/test/java/SaveGameTest.java]: fixed .. hope. I am still learning.

File: src/test/java/SaveGameTest.java (hunk 12 above)
Comment (szeliga71) [reply src/test/java/SaveGameTest.java]: as above

**Review by blazejprzyluski — APPROVED** [review]

**Pull request merged**